src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1501,22 +1514,28 @@ internal void UpdateModelValue(IEnumerable<Guid> modelGuids, string propertyName
 
         internal string GetStringRepOfWorkspace()
         {
-            // Create the xml document to write to.
-            var document = new XmlDocument();
-            document.CreateXmlDeclaration("1.0", null, null);
-            document.AppendChild(document.CreateElement("Workspace"));
-
-            //This is only used for computing relative offsets, it's not actually created
-            string virtualFileName = Path.Combine(Path.GetTempPath(), "DynamoTemp.dyn");
-            Utils.SetDocumentXmlPath(document, virtualFileName);
+            try
+            {
+                //Send the data in JSON format
+                //step 1: convert the model to json
+                var json = this.ToJson(null);
 
-            if (!PopulateXmlDocument(document))
-                return String.Empty;
+                //step2 : parse it as JObject
+                var jo = JObject.Parse(json);
 
-            //Now unset the temp file name again
-            Utils.SetDocumentXmlPath(document, null);
+                //step 2: raise the event to populate the view block
+                if (PopulateJSONWorkspace != null)
+                {
+                    json = PopulateJSONWorkspace(jo);
+                }
 
-            return document.OuterXml;
+                return json;
+            }
+            catch (Exception ex)
+            {
+                return ex.InnerException.ToString();
```

Comment:
Please could you JSON encode this, so that code that analyses the result doesn't have to guess whether it can decode it as JSON or not.

Something like JSON.encode({"exception" : ex.InnerException.ToString()})